### PR TITLE
Automatically upload published and draft posts with existing remote

### PR DESCRIPTION
**Pull request by leandroalonso**
Fixes a regression in #12324 

For tests instructions, please take a look in #12466 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [x] If it's feasible, I have added unit tests.

### CODE REVIEW

File: WordPress/Classes/Services/PostService+MarkAsFailedAndDraftIfNeeded.swift [hunk 1]
```diff
@@ -9,10 +9,6 @@
     ///     eventually be made private.
     ///
     func markAsFailedAndDraftIfNeeded(post: AbstractPost) {
-        guard post.remoteStatus != .failed, !post.hasRemote() else {
```
Comment (shiki) [inline WordPress/Classes/Services/PostService+MarkAsFailedAndDraftIfNeeded.swift]: I think we can keep the first part

```suggestion
    guard post.remoteStatus != .failed else {
```

File: WordPress/Classes/Services/PostService.m [hunk 2]
```diff
@@ -258,7 +258,7 @@ - (void)uploadPost:(AbstractPost *)post
         [self.managedObjectContext performBlock:^{
             Post *postInContext = (Post *)[self.managedObjectContext existingObjectWithID:postObjectID error:nil];
             if (postInContext) {
-                [self markAsFailedAndDraftIfNeededWithPost:postInContext];
+                [self markAsFailedAndDraftIfNeededWithPost:post];
```
Comment (shiki) [inline WordPress/Classes/Services/PostService.m]: ⚠️ What is the reason for this change? I thought it should be `postInContext` since `post` may be coming from a different thread.

Comment (leandroalonso) [reply WordPress/Classes/Services/PostService.m]: Oh sorry, I shouldn't have committed this one.

I did that when I was looking for the cause, reverted.

File: WordPress/Classes/Services/PostService+MarkAsFailedAndDraftIfNeeded.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/Services/PostService+MarkAsFailedAndDraftIfNeeded.swift]: Added back.

**Comment by leandroalonso** [comment]
@shiki This wasn't a logic directly related to #12466 but in the end it affected it.

Adding a test for this specific regression in there would imply an integration test, with multiple parts not being mocked. I would try to avoid that.

I added a new test case to ensure this will not happen anymore. I'm pretty comfortable with it by now, what do you think?

**Comment by shiki** [comment]
@leandroalonso Please merge it if you agree with my changes and the build passes. 🙂
